src/layers.rs

```diff
@@ -63,6 +63,44 @@ impl Embedding {
                 .var(name, &[num_embeddings, embedding_dim], init),
         )
     }
+
+    pub fn new_sinusoidal<'a>(
+        vs: impl Borrow<Path<'a>>,
+        name: &str,
+        num_embeddings: i64,
```

Comment:
The advantage of sinusoids is that we can give embeddings for every position, why limit to `num_embeddings`?

Reply:
To be fully compatible with other BERT models. I agree that it would be nice to create them on the fly, but then that would have a different model name.

Maybe we can define some traits soon so that we can abstract over the various types of embeddings.

Reply:
I guess one feature of the current implementation is that we can use sinusoid as initialization for learned embeddings.

Reply:
True. If we can process long enough sentences ;).

I'll soon make a dynamic-length implementation. We also do not need the type embeddings for our purposes. So, this could simply have word piece embeddings and generated position embeddings. I don't know what to call this data type though ;).